src/astro/utils/dependencies.py

```diff
@@ -0,0 +1,42 @@
+"""
+Copyright Astronomer, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+
+class MissingPackage(object):
+    def __init__(self, module_name):
+        self.module_name = module_name
+
+    def __getattr__(self, item):
+        raise RuntimeError(
+            f"Error loading the module {self.module_name},"
+            f" please make sure all the dependencies are installed"
```

Comment:
What do you think if we gave a more helpful message, telling the users how to install the missing packages?

Reply:
Yes, I agree, I think we can point out the command to run in error message itself.
Like then they can do `pip install astro-project[aws]` if package related to AWS was missing.

Reply:
That would be awesome, @utkarsharma2 !